lib/modelinstance.js

```diff
@@ -212,7 +212,9 @@ function _encodeValue(context, type, value, forceTyping, f) {
     if (!(value instanceof ModelInstance)) {
       throw new Error('Expected ' + f.name + ' type to be a ModelInstance.');
     }
-    if (type.name !== value.$.schema.name) {
+    // Values must match stated type names, unless the reference is to
+    // 'Mixed', then any reference will do.
+    if (type.name !== value.$.schema.name && (type.name !== 'Mixed')) {
```

Comment:
Can't fully explain this, but that won't work.  Sometimes 'Mixed' comes through as Schema.Mixed, other times it actually comes through as { type: 'Mixed' }.  While they're the same thing, you need it to always refer to the same function instance in order for the check as you suggest to work.